modules/nixos/fcitx5.nix

```diff
@@ -0,0 +1,19 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib) ctp mkIf;
+  cfg = config.i18n.inputMethod.fcitx5.catppuccin;
+  enable = cfg.enable && config.i18n.inputMethod.enabled == "fcitx5";
+in
+{
+  options.i18n.inputMethod.fcitx5.catppuccin = ctp.mkCatppuccinOpt "fcitx5";
+
+  config.i18n.inputMethod.fcitx5 = mkIf enable {
+    addons = [ pkgs.fcitx5-catppuccin ];
```

Comment:
we should be using our own `fcitx5` source like we do in the home-manager module